explore_data.R

```diff
@@ -0,0 +1,182 @@
+library(dplyr)
+library(tidyr)
+library(broom)
+library(feather)
+library(foreign)
+library(sas7bdat)
+
+##########
+# datasets - survey
+##########
+# survey_data folder
+# 1) "1987_2015_labour_force_survey"
+# 2) "2010_general_social_survey"
+# 3) "2010_imdb_tax_file"
+# 4) "2011_general_social_survey"
+# 5) "2012_general_social_survey"
+# 6) "2012_program_for_international_assessment_of_adult_comptencies"
+# 7)  "2013_general_social_survey"
+# 8) "2014_cananda_financial_capabilities_survey"
+# 9) "2014_employment_insurance_coverage_survey"
+# 10) "2014_general_social_survey"
+# 11) "2015_ontario_student_drug_use_and_health_survey"
+#12) "year_ontario_social_assistance_database"
+##########################################################################
+
+##########
+# datasets - NHS
+##########
+# nhs_data_folder
+# 1) "1987_2015_labour_force_survey"
+# 2) "2010_general_social_survey"
+# 3) "2010_imdb_tax_file"
+# 4) "2011_general_social_survey"
+# 5) "2012_general_social_survey"
+# 6) "2012_program_for_international_assessment_of_adult_comptencies"
+# 7)  "2013_general_social_survey"
+# 8) "2014_cananda_financial_capabilities_survey"
+# 9) "2014_employment_insurance_coverage_survey"
+# 10) "2014_general_social_survey"
+# 11) "2015_ontario_student_drug_use_and_health_survey"
+#12) "year_ontario_social_assistance_database"
+##########################################################################
+
+get_data <-
+  function(path_to_data) {
+
+    # get folders in path
+    data_folder <- list.files(path_to_data)
+    # get list to store objects
+    data_folders <- list()
+
+    for (i in 1:length(data_folder)) {
+      # nhs only has data, no sub folders
+      if (grepl('nhs', data_folder)) {
+        data_names <- data_folder
+      } else {
+        temp_folder <- data_folder[i]
+        # get list of data in given data folder
+        data_names <- list.files(paste(path_to_data, temp_folder, sep = '/'))
+      }
+      # get list to store data
+      data_list <- list()
+      for (j in 1:length(data_names)) {
+        temp_data <- data_names[j]
+        if(grepl('.csv', temp_data)){
+        temp_dat <- read.csv(file = paste(path_to_data,
```

Comment:
You don't necessarily need to change anything for this particular PR, but for future reference `readr::read_csv` is way faster than `base::read.csv` (which you're using here). Also, it defaults to `stringsAsFactors = F` (which is nice).